src/lib/components/DatePickerPanel.js

```diff
@@ -37,11 +38,7 @@ export default class DatePickerPanel extends React.Component {
     const { currentValue, confirmDisabled } = this.state;
     const disabledTimeFns = this.disabledTime();
 
-    const uiLocale = [browser.i18n.getUILanguage().replace('_', '-'), 'en-US'];
-    const dtf = new Intl.DateTimeFormat(uiLocale, {hour: 'numeric'});
-    const timeFormat = (dtf.resolvedOptions().hour12)
-      ? 'h:mm a'
-      : 'HH:MM';
+    const timeFormat = use12hFormat ? 'h:mm a' : 'HH:mm';
```

Comment:
My reasoning for keeping this part here is that we stopped using time formats all together in that file, since we're using Intl.DateTimeFormat, so this kind of time formatting belongs only to time picker.

I've also checked the component, and I'm afraid you will need to find a replacement if you want to fix the picker only showing 24h
https://github.com/react-component/time-picker/issues/27